src/test/fuzz/util.h

```diff
@@ -310,7 +311,8 @@ auto ConsumeNode(FuzzedDataProvider& fuzzed_data_provider, const std::optional<N
                                        addr_bind,
                                        addr_name,
                                        conn_type,
-                                       inbound_onion);
+                                       inbound_onion,
+                                       CNodeOptions{ .permission_flags = permission_flags });
```

Comment:
nit: here and below, could omit `CNodeOptions`, like done in `net.cpp`:

```suggestion
                                       { .permission_flags = permission_flags });
```

Reply:
Or maybe not: CI is upset:

```
./net.h:523:5: note:   no known conversion for argument 10 from ‘<brace-enclosed initializer list>’ to ‘CNodeOptions&&’
```

Reply:
Can't omit it here because `std::make_unique<CNode>(...)` can't resolve to a specific function without knowing what type `{ .permission_flags = .. }` is going to be. (Could omit it in the `else` clause, but figured it was better to have those two clauses be as similar as possible)

Reply:
Looks like you can't omit it when using `.foo = std::move(unique_ptr)` either, as gcc 8 gives errors. https://godbolt.org/z/rbn6csbax

Reply:
> Can't omit it here because `std::make_unique<CNode>(...)` can't resolve to a specific function without knowing what type `{ .permission_flags = .. }` is going to be.

This is ok, but an alternative is to switch from `make_unique<CNode>` to `new CNode`